src/pages/workspaces/workspace/Data.js

```diff
@@ -150,11 +157,16 @@ class WorkspaceData extends Component {
     )
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps, prevState) {
     StateHistory.update(_.pick(
-      ['workspaceEntities', 'selectedEntityType', 'selectedEntities', 'itemsPerPage', 'pageNumber'],
+      ['entityMetadata', 'selectedEntityType', 'selectedEntities', 'itemsPerPage', 'pageNumber'],
       this.state)
     )
+
+    if (this.state.itemsPerPage !== prevState.itemsPerPage || this.state.pageNumber !== prevState.pageNumber) {
+      this.setState({ isDataLoaded: false })
```

Comment:
It might be cleaner to manage this state centrally in `loadEntities`, tied directly to the ajax call.